### PR TITLE
feat: [PL-23547]: UI for azure blob sm

### PR DESCRIPTION
**Pull request by tiwaritejas**
##### Screenshots:

You can use the following comments to re-trigger PR Checks

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier - `fix prettier`

Screenshots- 

Screenshot 2022-04-25 at 11 59 59 AM
Screenshot 2022-04-25 at 12 00 04 PM
Screenshot 2022-04-25 at 12 00 09 PM
Screenshot 2022-04-25 at 12 00 14 PM
Screenshot 2022-04-25 at 12 00 24 PM
Screenshot 2022-04-25 at 12 04 14 PM

### CODE REVIEW

**Comment by meenakshi-harness** [comment]
dockerize

**Comment by tiwaritejas** [comment]
dockerize

File: src/modules/35-connectors/strings/strings.en.yaml [hunk 1]
```diff
@@ -786,3 +787,14 @@ ceGcp:
...
+  labels:
+    connectionString: Azure Storage Connection String
+    containerURL: Azure Storage Container URL
+    keyName: Azure Key Name
+    keyId: Azure Key ID
+  validation:
+    connectionStringIsRequired: Azure Storage Connection String is required
+    containerURLIsRequired: Azure Storage Container URL is required
+    keyNameIsRequired: Azure Key Name is required
+    keyIdIsRequired: Azure Key ID is required
```
Comment (meenakshi-harness) [inline src/modules/35-connectors/strings/strings.en.yaml:800]: Are these still required?

File: src/modules/35-connectors/components/CreateConnector/CreateAzureBlobConnector/views/AzureBlobForm.tsx [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+      </Text>
+      <Formik<AzureBlobFormData>
+        formName="azureBlobForm"
+        enableReinitialize
+        initialValues={{ ...initialValues, ...prevStepData }}
+        validationSchema={Yup.object().shape({
+          containerURL: Yup.string().required(getString('connectors.azureBlob.validation.containerURLIsRequired')),
+          clientId: Yup.string().required(getString('common.validation.clientIdIsRequired')),
+          tenantId: Yup.string().required(getString('connectors.azureKeyVault.validation.tenantId')),
+          secretKey: Yup.string().when('vaultName', {
```
Comment (meenakshi-harness) [inline src/modules/35-connectors/components/CreateConnector/CreateAzureBlobConnector/views/AzureBlobForm.tsx:75]: why checking this validation based on vaultName?

File: src/modules/35-connectors/components/CreateConnector/CreateAzureBlobConnector/CreateAzureBlobConnector.tsx [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+      <VerifyOutOfClusterDelegate
+        name={getString('connectors.stepThreeName')}
+        connectorInfo={props.connectorInfo}
+        isStep
+        onClose={onClose}
+        isLastStep={true}
+        type={Connectors.AZURE_KEY_VAULT}
+        stepIndex={GIT_TESTCONNECTION_STEP_INDEX}
+      />
+    </StepWizard>
```
Comment (meenakshi-harness) [inline src/modules/35-connectors/components/CreateConnector/CreateAzureBlobConnector/CreateAzureBlobConnector.tsx:73]: Can you add screenshots for this component?

File: src/modules/35-connectors/pages/connectors/utils/ConnectorUtils.ts [hunk 4]
```diff
@@ -1528,13 +1548,34 @@ export const setupAzureKeyVaultFormData = async (
     secretKey: secretKey || undefined,
     tenantId: connectorInfoSpec?.tenantId || undefined,
     subscription: connectorInfoSpec?.subscription || undefined,
+    connectionString: connectorInfoSpec?.connectionString || undefined,
+    containerName: connectorInfoSpec?.containerName || undefined,
```
Comment (meenakshi-harness) [inline src/modules/35-connectors/pages/connectors/utils/ConnectorUtils.ts:1552]: These are not needed now?

File: src/modules/35-connectors/pages/connectors/utils/ConnectorUtils.ts [hunk 5]
```diff
@@ -1528,13 +1548,34 @@ export const setupAzureKeyVaultFormData = async (
...
     default: connectorInfoSpec?.default || false
   }
 }
 
 export const setupAzureKeyVaultNameFormData = async (connectorInfo: ConnectorInfoDTO): Promise<FormData> => {
   return {
-    vaultName: connectorInfo?.spec?.vaultName
+    vaultName: connectorInfo?.spec?.vaultName,
+    keyName: connectorInfo?.spec?.keyName,
+    keyId: connectorInfo?.spec?.keyId
```
Comment (meenakshi-harness) [inline src/modules/35-connectors/pages/connectors/utils/ConnectorUtils.ts:1578]: not needed

File: src/modules/35-connectors/pages/connectors/views/savedDetailsView/SavedConnectorDetails.tsx [hunk 6]
```diff
@@ -528,6 +529,24 @@ const getAzureKeyVaultSchema = (connector: ConnectorInfoDTO): Array<ActivityDeta
...
+      value: data.clientId
+    },
+    {
+      label: 'connectors.azureKeyVault.labels.tenantId',
+      value: data.tenantId
+    },
+    {
+      label: 'connectors.azureBlob.labels.containerURL',
+      value: data.containerURL
+    }
```
Comment (meenakshi-harness) [inline src/modules/35-connectors/pages/connectors/views/savedDetailsView/SavedConnectorDetails.tsx:546]: add Default  as well.

File: src/modules/35-connectors/components/CreateConnector/CreateAzureBlobConnector/views/AzureBlobForm.tsx (hunk 2 above)
Comment (tiwaritejas) [reply src/modules/35-connectors/components/CreateConnector/CreateAzureBlobConnector/views/AzureBlobForm.tsx:75]: Removed

**Comment by tiwaritejas** [comment]
dockerize

**Comment by tiwaritejas** [comment]
retrigger titlecheck

**Comment by tiwaritejas** [comment]
dockerize

**Comment by tiwaritejas** [comment]
retrigger standards

**Comment by tiwaritejas** [comment]
triggegr prettier

File: src/modules/35-connectors/components/CreateConnector/CreateAzureBlobConnector/views/AzureBlobForm.tsx [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+  const [loadingFormData, setLoadingFormData] = useState(isEditMode)
+
+  React.useEffect(() => {
+    if (isEditMode && connectorInfo) {
+      setupAzureBlobFormData(connectorInfo, accountId).then(data => {
+        setInitialValues(data as AzureBlobFormData)
+        setLoadingFormData(false)
+      })
+    }
+  }, [isEditMode, connectorInfo])
```
Comment (diptiman-adak) [inline src/modules/35-connectors/components/CreateConnector/CreateAzureBlobConnector/views/AzureBlobForm.tsx:50]: Codacy has a fix for the issue: [React Hook React.useEffect has a missing dependency: 'accountId'. Either include it or remove the dependency array.](https://app.codacy.com/gh/harness/harness-core-ui/pullRequest?prid=9232296)
```suggestion
  }, [isEditMode, connectorInfo, accountId])
```

Comment (diptiman-adak) [inline src/modules/35-connectors/components/CreateConnector/CreateAzureBlobConnector/views/AzureBlobForm.tsx:50]: Codacy has a fix for the issue: [React Hook React.useEffect has a missing dependency: 'accountId'. Either include it or remove the dependency array.](https://app.codacy.com/gh/harness/harness-core-ui/pullRequest?prid=9232296)
```suggestion
  }, [isEditMode, connectorInfo, accountId])
```

**Comment by meenakshi-harness** [comment]
trigger cypress

**Review by meenakshi-harness — DISMISSED** [review]

**Review by mayankVermaHarness — APPROVED** [review]

File: src/modules/35-connectors/components/CreateConnector/CreateAzureBlobConnector/__tests__/CreateAzureBlobConnector.test.tsx [hunk 8]
```diff
@@ -0,0 +1,277 @@
...
+    const secret = await findByText(modal!, 'mockSecret')
+    await act(async () => {
+      fireEvent.click(secret)
+    })
+    const applyBtn = await waitFor(() => findByText(modal!, 'entityReference.apply'))
+    await act(async () => {
+      fireEvent.click(applyBtn)
+    })
+
+    expect(container).toMatchSnapshot()
```
Comment (abhinavrastogi-harness) [inline src/modules/35-connectors/components/CreateConnector/CreateAzureBlobConnector/__tests__/CreateAzureBlobConnector.test.tsx:175]: do we need so many snapshot checks?
can we use assertions instead? snapshots are generally very flaky

File: src/modules/35-connectors/components/CreateConnector/CreateAzureBlobConnector/__tests__/CreateAzureBlobConnector.test.tsx [hunk 9]
```diff
@@ -0,0 +1,277 @@
...
+
+    await act(async () => {
```
Comment (abhinavrastogi-harness) [inline src/modules/35-connectors/components/CreateConnector/CreateAzureBlobConnector/__tests__/CreateAzureBlobConnector.test.tsx:177]: ayns-await is usually not required with `act()`

File: src/modules/35-connectors/components/CreateConnector/CreateAzureBlobConnector/views/AzureBlobForm.tsx [hunk 10]
```diff
@@ -0,0 +1,98 @@
...
+import type { StepDetailsProps, ConnectorDetailsProps } from '@connectors/interfaces/ConnectorInterface'
+import AzureBlobFormFields from './AzureBlobFormFields'
+import css from '../CreateAzureBlobConnector.module.scss'
+
+export interface AzureBlobFormData {
+  clientId?: string
+  secretKey?: SecretReference
+  tenantId?: string
+  containerURL?: string
+  default?: boolean
```
Comment (abhinavrastogi-harness) [inline src/modules/35-connectors/components/CreateConnector/CreateAzureBlobConnector/views/AzureBlobForm.tsx:25]: how is everything optional in this form?

File: src/modules/35-connectors/components/CreateConnector/CreateAzureBlobConnector/views/AzureBlobForm.tsx [hunk 11]
```diff
@@ -0,0 +1,98 @@
...
+    <Container padding={{ top: 'medium' }} width="64%">
+      <Text font={{ variation: FontVariation.H3 }} padding={{ bottom: 'xlarge' }}>
+        {getString('details')}
+      </Text>
+      <Formik<AzureBlobFormData>
+        formName="azureBlobForm"
+        enableReinitialize
+        initialValues={{ ...initialValues, ...prevStepData }}
+        validationSchema={Yup.object().shape({
+          containerURL: Yup.string().required(getString('connectors.azureBlob.validation.containerURLIsRequired')),
```
Comment (abhinavrastogi-harness) [inline src/modules/35-connectors/components/CreateConnector/CreateAzureBlobConnector/views/AzureBlobForm.tsx:62]: no URL validation on this field?

Comment (tiwaritejas) [reply src/modules/35-connectors/components/CreateConnector/CreateAzureBlobConnector/views/AzureBlobForm.tsx:62]: This URL is more like an identifier for the container and not some URL that we ping. We will add this validation later if needed as customers can have different format for the URL for different Azure environments.

File: src/modules/35-connectors/components/CreateConnector/CreateAzureBlobConnector/__tests__/CreateAzureBlobConnector.test.tsx (hunk 8 above)
Comment (tiwaritejas) [reply src/modules/35-connectors/components/CreateConnector/CreateAzureBlobConnector/__tests__/CreateAzureBlobConnector.test.tsx:173]: Reduced the snapshots and awaiting appropriately before expect() to reduce flakiness. Will add more assertions explicitly later.

**Review by abhinavrastogi-harness — APPROVED** [review]

**Review by meenakshi-harness — APPROVED** [review]

**Review by abhinavrastogi-harness — APPROVED** [review]

**Pull request closed without merging**